shell/browser/native_window_mac.mm

```diff
@@ -1345,15 +1345,37 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   original_frame_ = [window_ frame];
 }
 
-void NativeWindowMac::SetVibrancy(const std::string& type) {
+void NativeWindowMac::SetVibrancy(const std::string& type,
+                                  bool animate,
+                                  int duration) {
   NSVisualEffectView* vibrantView = [window_ vibrantView];
 
   if (type.empty()) {
     if (vibrantView == nil)
       return;
 
-    [vibrantView removeFromSuperview];
-    [window_ setVibrantView:nil];
+    vibrancy_type_ = type;
+
+    if (animate) {
+      __weak auto weak_delegate = window_delegate_.get();
+      [NSAnimationContext
+          runAnimationGroup:^(NSAnimationContext* context) {
+            context.duration = duration / 1000.0f;
+            vibrantView.animator.alphaValue = 0.0;
+          }
```

Comment:
What happens if this is called multiple times in parallel? e.g.

```js
win.setVibrancy(false, { animated: true })
setTimeout(() => {
  win.setVibrancy(true, { animated: true })
}, 10)
```

does the 2nd animation cancel the first one? if not, the code for removing the vibrancy view below is incorrect, as it will remove the vibrancy after the window's vibrancy has been set to `true`.

Reply:
yes the second animation removes the first one. so for example this snippet:

``` js
win.setVibrancy(null, {
  animate: true,
  animationDuration: 5000
});

setTimeout(() => {
  win.setVibrancy('hud', {
    animate: true,
    animationDuration: 500
  });
}, 1000);
```

produces the result below, where you can see the slow first remove animation is replaced by a fast blur animation.

https://user-images.githubusercontent.com/1059699/206922596-20c57909-06c5-48eb-8f4e-eaf26c21b42c.mov